src/kili/entrypoints/mutations/project/__init__.py

```diff
@@ -444,3 +446,58 @@ def update_project_anonymization(
 
         result = self.graphql_client.execute(GQL_PROJECT_UPDATE_ANONYMIZATION, variables)
         return format_result("data", result)
+
+    @typechecked
+    def import_text_from_csv(
+        self,
+        csv_file: str,
+        content_column: str,
+        unique_categories: Optional[List[str]] = None,
+        category_column: Optional[str] = None,
+        external_id_column: Optional[str] = None,
+        csv_separator: str = ",",
+        label_type: LabelType = "DEFAULT",
+        project_id: Optional[str] = None,
+        project_title: Optional[str] = None,
+        project_description: Optional[str] = None,
+    ) -> Dict[Literal["id"], str]:
+        """Import text assets and labels from a csv file.
+
+        This method imports assets and eventually assigns a single category to each asset (if
+        `category_column` is provided).
+
+        The csv file should contain a header with the column names.
+
+        Args:
+            csv_file: Path to the csv file.
+            content_column: Name of the column containing the content of the assets.
+            unique_categories: List of the categories an asset can belong to.
+                This argument is used for project creation if `project_id` is None.
+                If not provided, the categories will be inferred from the csv `category_column`.
+            category_column: Name of the column containing the category of the assets.
+                If None, the assets will be imported without labels.
+            external_id_column: Name of the column containing the external id of the assets.
+                If None, the external id will be set to the row number.
+            label_type: Type of the labels to create. Only used if `category_column` is provided.
+            csv_separator: Separator used in the csv file.
+            project_id: Identifier of the project. If None, a text project is created.
+            project_title: Title of the project. Only required if `project_id` is None.
+            project_description: Description of the project. Only required if `project_id` is None.
+
+        Returns:
+            A dictionary with the project `id`.
+        """
+        project_id = import_text_assets_labels_from_csv(
```

Comment:
product side question:
I really like the fact to be able to import labels in the same time. We wanted to add this feature in append_many_to_dataset in the SDK v3.
I never thought of the possibility to create a project in the same time, I find it also interesting to be able to create a simple classification project if not project_id is provided.
My only problem is that it participates in the SDK heterogeneity, and thus onboarding issues for someone discovering the SDK. 
We have only one use case where we have a function for project creation, asset import and label import and not others. 

Plus the fact that the function does 3 things make it hard to take in hands (nothing very though, I think that you handled it very well but it still have some complexity)

Why didn't we added a parameter `from_csv` in append_many_to_dataset and crashing an error if the project is not a text project ?
You maybe already had the discussion the Pierre of the product team but I am curious to know where this idea of handling everything with this function come from